source/MaterialXCore/Interface.cpp

```diff
@@ -43,6 +43,11 @@ const std::unordered_map<string, size_t> PortElement::CHANNELS_PATTERN_LENGTH =
     { "vector4", 4 }
 };
 
+auto VALUE_ELEMENT_COMPARE = [](ValueElementPtr lhs, ValueElementPtr rhs)
```

Comment:
I changed this to a lambda, but I think it still needs to be global as otherwise I would need to repeat it 3 times.

Reply:
Maybe you could define a type for the set, including the custom lambda for less operator. If it's only private in this file put it in a anonymous namespace. For example:

```
namespace
{
    auto ValueElemmentLess = [](ValueElementPtr lhs, ValueElementPtr rhs)
    {
        return lhs->getName() < rhs->getName();
    };

    using ValueElementSet = std::set<ValueElementPtr, decltype(ValueElemmentLess)>;
}

```